src/typical.jl

```diff
@@ -79,9 +80,9 @@ function typify(refgrid, model_formula::FormulaTerm,
     return _replace(matrix_term, typical_terms)
 end
 
-_replace(matrix_term::MatrixTerm, typicals::Dict) = MatrixTerm(_replace.(matrix_term.terms, Ref(typicals)))
+_replace(matrix_term::MatrixTerm, typicals::Dict) = MatrixTerm(_replace.(terms(matrix_term), Ref(typicals)))
```

Comment:
I think this should fix the current method error:
```suggestion
function _replace(matrix_term::MatrixTerm, typicals::Dict)
    ts = _replace.(terms(matrix_term), Ref(typicals))
    return MatrixTerm(Tuple(ts))
end

```
Do note that for `mt::MatrixTerm`, `terms(mt)` is actually not the same as `mt.terms` but I _think_ this should work anyway.

Reply:
@ararslan Since `MatrixTerm` is a concrete type, we could probably just undo the change in this line anyway? The problem was with the `terms` property in other subtypes of `AbstractTerm`